pkg/depository/create.go

```diff
@@ -17,18 +17,243 @@ limitations under the License.
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
 
+	"github.com/bestchains/bestchains-contracts/library"
 	"github.com/spf13/cobra"
 )
 
 func NewCreateDepositoryCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use: "depository NAME [args]",
-		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Println("create depository")
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// Get depot info from flags
+			n, err := cmd.Flags().GetString("name")
+			if err != nil {
+				return err
+			}
+			t, err := cmd.Flags().GetString("contentType")
+			if err != nil {
+				return err
+			}
+			id, err := cmd.Flags().GetString("contentID")
+			if err != nil {
+				return err
+			}
+			p, err := cmd.Flags().GetString("platform")
+			if err != nil {
+				return err
+			}
+			acc, err := cmd.Flags().GetString("wallet-account")
+			if err != nil {
+				return err
+			}
+			key, err := cmd.Flags().GetString("privateKey")
+			if err != nil {
+				return err
+			}
+			accGen, err := cmd.Flags().GetBool("wallet-account-generate")
+			if err != nil {
+				return err
+			}
+			untrusted, err := cmd.Flags().GetBool("untrusted")
+			if err != nil {
+				return err
+			}
+
+			// FIXME: the host should be read from the configuration file.
+			host, _ := cmd.Flags().GetString("host")
+			if host == "" {
+				return fmt.Errorf("no host provided")
+			}
+
+			valueBase64 := generateValueDepotBase64(n, t, id, p)
+
+			if !accGen {
+				// FIXME: key auth to be done, bypassed temporally
+				// key auth
+				var privKey ecdsa.PrivateKey
+				err = json.Unmarshal([]byte(key), &privKey)
+				if err != nil {
+					return fmt.Errorf("privKey unmarshal error: " + err.Error())
+				}
+				if !keyAuth(privKey) {
+					return fmt.Errorf("key Auth failed")
+				}
+				// get nonce
+				non := getNonce(acc)
+				// generate msg
+				msgBase64 := generateMessageBase64(non, privKey, []byte(valueBase64))
+				// POST PutValue request
+				postValue := url.Values{}
+				postValue.Add("message", msgBase64)
+				postValue.Add("value", valueBase64)
+
+				// -> http://localhost/basic/putValue
+				host = fmt.Sprintf("%s%s", host, common.CreateDepository)
+				body := bytes.NewBuffer([]byte(postValue.Encode()))
+				resp, err := uhttp.Do(host, http.MethodPost, map[string]string{
+					"Content-Type": "application/x-www-form-urlencoded",
+				}, body.Bytes())
+				if err != nil {
+					return err
+				}
+				fmt.Print(string(resp))
+				return nil
+			} else {
+				if untrusted {
+					fmt.Print("putting untrusted value...\n")
+					postValue := url.Values{}
+					postValue.Add("value", valueBase64)
+
+					// -> http://localhost/basic/putUntrustValue
+					host = fmt.Sprintf("%s%s", host, common.CreateUntrustedDepository)
+					body := bytes.NewBuffer([]byte(postValue.Encode()))
+					resp, err := uhttp.Do(host, http.MethodPost, map[string]string{
+						"Content-Type": "application/x-www-form-urlencoded",
+					}, body.Bytes())
```

Comment:
If the function argument is a `[]byte`, use `[]byte(postValue.Encode())` directly.